pkg/arvo/sys/hoon.hoon

```diff
@@ -11457,6 +11456,24 @@
     %-  ~(play ut p.vax)
     [%wtgr [%wtts [%leaf %tas -.q.vax] [%& 2]~] [%$ 1]]
   (~(fuse ut p.vax) [%cell %noun %noun])
+::  +shot: deferred +slam
+::
+++  shot
+  |=  [gat=(trap vase) sam=(trap vase)]
+  ^-  (trap vase)
+  =/  [typ=type gen=hoon]
+    :-  [%cell p:$:gat p:$:sam]
+    [%cnsg [%$ ~] [%$ 2] [%$ 3] ~]
+  =+  gun=(~(mint ut typ) %noun gen)
+  |.  ~+
+  [p.gun (slum q:$:gat q:$:sam)]
+::  +slew: deferred +slop
+::
+++  slew
+  |=  [hed=(trap vase) tal=(trap vase)]
+  ^-  (trap vase)
+  |.  ~+
+  [[%cell p:$:hed p:$:tal] [q:$:hed q:$:tal]]
```

Comment:
This needs a minimized subject outside the `|.` (with `=>  +<` or equivalent). Inside, you should pin `$:hed` and `$:tal` to avoid unnecessarily reevaluating them.